src/smart-components/Thread/components/ParentMessageInfo/index.tsx

```diff
@@ -0,0 +1,193 @@
+import React, { useMemo, useState } from 'react';
+import format from 'date-fns/format';
+import { GroupChannel } from '@sendbird/chat/groupChannel';
+import { FileMessage, UserMessage } from '@sendbird/chat/message';
+
+import './index.scss';
+
+import Avatar from '../../../../ui/Avatar';
+import Label, { LabelTypography, LabelColors } from '../../../../ui/Label';
+import {
+  getSenderName,
+  getUIKitMessageType,
+  getUIKitMessageTypes,
+  isOGMessage,
+  isTextMessage,
+  isThumbnailMessage,
+} from '../../../../utils';
+import { useLocalization } from '../../../../lib/LocalizationContext';
+import TextMessageItemBody from '../../../../ui/TextMessageItemBody';
+import OGMessageItemBody from '../../../../ui/OGMessageItemBody';
+import FileMessageItemBody from '../../../../ui/FileMessageItemBody';
+import ThumbnailMessageItemBody from '../../../../ui/ThumbnailMessageItemBody';
+import UnknownMessageItemBody from '../../../../ui/UnknownMessageItemBody';
+import EmojiReactions from '../../../../ui/EmojiReactions';
+import useSendbirdStateContext from '../../../../hooks/useSendbirdStateContext';
+import { useThreadContext } from '../../context/ThreadProvider';
+import MessageItemMenu from '../../../../ui/MessageItemMenu';
+import RemoveMessage from '../RemoveMessageModal';
+import MessageItemReactionMenu from '../../../../ui/MessageItemReactionMenu';
+
+interface ParentMessageInfoProps {
+  className?: string;
+  channel: GroupChannel;
+  message: UserMessage | FileMessage;
+}
+
+export default function ParentMessageInfo({
+  className,
+  channel,
+  message,
+}: ParentMessageInfoProps): React.ReactElement {
+  const { stores, config } = useSendbirdStateContext();
+  const {
+    isMentionEnabled,
+    isReactionEnabled,
+    replyType,
+  } = config;
+  const userId = stores?.userStore?.user?.userId;
+  const { dateLocale } = useLocalization?.();
+  const messageTypes = getUIKitMessageTypes?.();
+  const {
+    emojiContainer,
+    nicknamesMap,
+    toggleReaction,
+    currentChannel,
+  } = useThreadContext();
+
+  const [showRemove, setShowRemove] = useState(false);
+  const [supposedHover, setSupposedHover] = useState(false);
+  const usingReaction = isReactionEnabled && !currentChannel?.isSuper && !currentChannel?.isBroadcast
+
+  const MemorizedEmojiReactions = useMemo(() => {
```

Comment:
Dont need to memoize these components yet.. but its okay, also donot need to rewrite 😄